frappe/public/js/frappe/form/controls/date.js

```diff
@@ -141,13 +142,21 @@ frappe.ui.form.ControlDate = class ControlDate extends frappe.ui.form.ControlDat
 			}
 		});
 	}
+	/** @param {string | null} value */
 	parse(value) {
-		if (value) {
-			if (value == "Invalid date") {
-				return "";
-			}
-			return frappe.datetime.user_to_str(value, false, true);
+		value = typeof value === "string" ? value : "";
+
+		if (["today", "now"].includes(value.toLowerCase())) {
```

Comment:
Accepting “now“ even for ControlDate. Good.

But as below, let’s add a `as_obj` param to our helper function get_now_date() and use it consistently.